tutorials/define_and_compile_model_webgl.py

```diff
@@ -0,0 +1,206 @@
+"""
+Quick Start - End-to-End Tutorial for NNVM/TVM Pipeline for OpenGL and WebGL
+============================================================================
+**Author**: `Zhixun Tan <https://github.com/phisiart>`_
+
+This example shows how to build a neural network with NNVM python frontend and
+generate runtime library for WebGL running in a browser with TVM. (Thanks to
+Tianqi's `tutorial for cuda <http://nnvm.tvmlang.org/tutorials/get_started.html>`_ and
+Ziheng's `tutorial for Raspberry Pi <http://nnvm.tvmlang.org/tutorials/deploy_model_on_rasp.html>`_)
+To run this notebook, you need to install tvm and nnvm following
+`these instructions <https://github.com/dmlc/nnvm/blob/master/docs/how_to/install.md>`_.
+Notice that you need to build tvm with OpenGL.
+"""
+
+######################################################################
+# Overview for Supported Hardware Backend of TVM
+# -----------------------------
+# The image below shows hardware backend currently supported by TVM:
+#
+# .. image:: https://github.com/dmlc/web-data/raw/master/tvm/tutorial/tvm_support_list.png
+#      :align: center
+#      :scale: 100%
+#
+# In this tutorial, we'll choose OpenGL as the target backend.
+# To begin with, let's import NNVM and TVM.
+import tvm
+import nnvm.compiler
+import nnvm.testing
+
+
+######################################################################
+# Define Neural Network in NNVM
+# -----------------------------
+# First, let's define a neural network with nnvm python frontend.
+# For simplicity, we'll use pre-defined resnet network in NNVM.
+# Parameters are initialized with Xavier initializer.
+# NNVM also supports other model formats such as MXNet, CoreML and ONNX.
+#
+# In this tutorial, we assume we will do inference on our device
+# and the batch size is set to be 1. Input images are RGB color
+# images of size 16 * 16. We can call the :any:`nnvm.symbol.debug_str`
+# to show the network structure.
+
+batch_size = 1
+num_class = 2
```

Comment:
Note: I use a smaller setting here. When using the original model, the js module crashes complaining "memory too small".